includes/class-newspack-ui.php

```diff
@@ -64,13 +66,130 @@ public static function colors_css_wrap() {
 		wp_add_inline_style( 'newspack-ui', $custom_css );
 	}
 
+	/**
+	 * Generate markup for a Newspack UI modal.
+	 *
+	 * @param array $args {
+	 *     Arguments for building the modal.
+	 *     @type string $id The modal ID.
+	 *     @type string $title The modal title.
+	 *     @type string $content The modal content HTML.
+	 *     @type string $footer The modal footer HTML.
+	 *     @type string $form The form method to use. If given, modal content and action buttons will be wrapped in a form element.
+	 *     @type array $actions {
+	 *         @type string $label The button label.
+	 *         @type string $type The button type.
+	 *         @type string $action The action to perform when the button is clicked. Currently only: 'close' to close the modal.
+	 *         @type string $url The URL to navigate to when the button is clicked.
+	 *         @type array {
+	 *               $fetch Data to trigger a REST API request.
+	 *               @type string $url The URL to send the request to.
+	 *               @type string $method The HTTP method to use.
+	 *               @type string $nonce The nonce to use for the request.
+	 *               @type array $body The body to send with a POST request.
+	 *               @type array $params Parameters to send with a GET request.
+	 *               @type string $next The ID for another modal to open after the action is completed.
+	 *         }
+	 *     }
+	 * }
+	 */
+	public static function generate_modal( $args ) {
```

Comment:
I think not for this PR, but we could benefit from making this a class. Perhaps the next iteration on the project that uses the modal can expand on this. WDYT?

Reply:
Yeah, this can probably be expanded upon into a more robust class where you pass arguments via the class constructor. I definitely welcome any improvements!